pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -305,23 +305,36 @@ func maybeRetryDownload(ctx *downloaderContext,
 			status.Key())
 		return
 	}
-	status.LastErr = ""
-	status.LastErrTime = time.Time{}
-	status.RetryCount += 1
+	status.ClearErrorInfo()
 	// XXX do we need to adjust reservedspace??
 
 	dst, errStr := lookupDatastoreConfig(ctx, config.DatastoreID, config.Name)
 	if dst == nil {
-		status.LastErr = errStr
-		status.LastErrTime = time.Now()
+		status.SetErrorInfo(errStr)
 		publishDownloaderStatus(ctx, status)
 		return
 	}
+
+	// try to reserve storage
+	// must be released on error
+	kb := types.RoundupToKB(config.Size)
+	if !tryReserveSpace(ctx, status, kb) {
+		errStr := fmt.Sprintf("Would exceed remaining space. "+
+			"SizeOfAppImage: %d, RemainingSpace: %d\n",
+			kb, ctx.globalStatus.RemainingSpace)
+		log.Errorln(errStr)
```

Comment:
There is an other Errorf below.. Better to combine into one statement?

Reply:
done.